### ISSUE TITLE
some questions about loading checkpoints

### ISSUE THREAD

**Issue by aiyolo** [PyTorchLightning/pytorch-lightning#3387]
What I have tried:
- defined a Lighting Module class: `model=LitModel(hparams)`, 
- then I trained the model and got a checkpoint; 
- since I want to visualize the features of the trainset, I write a another file and loading the checkpoint, `model = LitModel(hparams).load_from_checkpoint('lightning_logs/version_90/checkpoints/epoch=119.ckpt')`

My questions are:
- I have to add a line `model.eval()` before `model(samples)`, otherwise, the accuracy is very low! however, to my best knowledge, in common pytorch code, the accuracy of the trainset has no matter with `model.eval()`, **so why `model.eval()` take effects?**
- I tried to put my model onto gpu`model = LitModel(hparams).load_from_checkpoint('lightning_logs/version_90/checkpoints/epoch=119.ckpt').cuda()`, the model takes up about 460M/11G, however, when running into `model(samples.cuda())`, my gpu memory is out of use! **so, can anyone show me the correct way to visualize features using the checkpoint of Light Module?**

**Comment by rohitgr7** [comment]
> my gpu memory is out of use!

it will go up due to activations calculated in the forward pass. I think you can use torch.no_grad() to lower memory consumption.

> model.eval()

If changes the behavior of batchnorm and dropout layers.

**Comment by aiyolo** [comment]
but it still can't explain why the gpu memory is enough for training a model in the training phase while getting exhausted in the feature visualizing stage, they are using the same dataloader.

**Comment by rohitgr7** [comment]
hmm... this should not happen. During inference it's just a simple pytorch model, nothing fancy. Is it possible for you to share a colab notebook??
